components/automate-ui/src/app/entities/environments/environment.model.ts

```diff
@@ -1,3 +1,38 @@
+import { merge } from 'lodash';
+
 export interface Environment {
   name: string;
+  chef_type: string;
+  description: string;
+  json_class: string;
+  default_attributes: string;
+  override_attributes: string;
+  run_list: string[];
+  cookbook_versions: CookbookVersion[];
+}
+
+export interface CookbookVersion {
+  name: string;
+  version: string;
+}
+
+export class EnvironmentAttributes {
+  default_attributes: string;
+  override_attributes: string;
+  all: string;
+  constructor(resp: RespEnvironmentAttributes) {
+    this.default_attributes = resp.default_attributes &&
+      JSON.parse(resp.default_attributes) || {};
+    this.override_attributes = resp.override_attributes &&
+      JSON.parse(resp.override_attributes) || {};
+    this.all = merge(
+      {},
+      this.default_attributes,
+      this.override_attributes);
+  }
```

Comment:
Both `default_attributes` and `override_attributes` are purportedly strings and it does not make sense to use those in a `merge` operation. I suspect it is the declarations of those that are wrong, since `JSON.parse` returns an object not a string. Assuming that is the case and you need to fix the declarations, I think you can even obviate the need for the `merge` operator and simplify with:
```suggestion
    this.all = {...this.default_attributes, ...this.override_attributes};
  }
```
Bonus points for making the same fix in other places that this pattern is used. 😁

Reply:
Oh, I should have mentioned: the JavaScript spread operator (`...`) does only _shallow_ cloning. If you need deeper, then you would likely need to stay with `merge`.

Reply:
I tried but I am getting `Spread types may only be created from object types.` so, for now, I am staying with the `merge`.

Reply:
You have missed my major concern here: there is something wrong with the types.
According to these lines:
https://github.com/chef/automate/blob/ee6ec4188a6d2b2a65ebd20964eaa726f43d74f3/components/automate-ui/src/app/entities/environments/environment.model.ts#L24-L27
...you are storing _objects_ in `default_attributes` and `override_attributes`. But those are declared as _strings_.

Reply:
I have changed to only `Object`

Reply:
@vinay033 The line you changed was only one small part of the concern I was attempting to outline. As I stated above in environment.model.ts, "you are storing _objects_ in default_attributes and override_attributes. But those are declared as _strings_." 

➡️ Please review the entire file of environment.model.ts to correct that type mismatch.

Once that is done, make sure that any consumers (like L112 environment-details.component.ts that I showed as just one example) are correct as well.

Also, once you have done that, I still suggest the spread operator can then be used instead of the merge function (assuming a shallow copy is sufficient for your needs).

Thanks!

Reply:
@msorens , I checked did some R&D on this, actually, we don't know which kind of data we are receiving in API response(maybe string or maybe object) so we are not exactly sure about the type. for reference, you can check this as well. type.https://github.com/chef/automate/blob/6b3840d29126d224a816ac118666193cfb3fb131/components/automate-ui/src/app/page-components/attributes/attributes.component.ts#L66 so we need to use `string | Object` or `any`.

Reply:
In this bit of code...
```
export class EnvironmentAttributes {
  default_attributes: string; // (1)
  override_attributes: string;
  all: string;
  constructor(resp: RespEnvironmentAttributes) {
    this.default_attributes = // (2)
      (resp.default_attributes && JSON.parse(resp.default_attributes)) || {};
    this.override_attributes =
      (resp.override_attributes && JSON.parse(resp.override_attributes)) || {};
```
Notice that `default_attributes` is declared as a string (1). You are then assigning an object (in all cases) to that property (2). That is the type mismatch I have been referring to.

Reply:
@msorens I have changed to `string` only to fix type mismatch.

Reply:
I was going to take a look but I need instructions on how to arrive at the pages shown in your screen shots above. Please update your PR preamble with necessary details so I can exercise the code. Thanks!

Reply:
By the way, looks like this issue is not localized to `environments`. Looks like it was copied from `roles`. So if you want to merge this as is and then address the issue for both roles and environments in a separate PR, that is fine. But if you choose to do so, please add a new card so this issue gets resolved as soon as we can.

Reply:
![nvaigate](https://user-images.githubusercontent.com/12297653/82410737-0ea5c580-9a8e-11ea-9795-f681b1c2f488.png)

Reply:
roles and environment both code is added by me and I only used the pre-existing `Json tree`  component of automate UI lib, and I was mentioned the reference files. so we can fix this on a separate card.
This code is not only in this PR, `Json tree` is also using in `clients run` page ==> `attributes.component.ts` so I referred the same for now but we need to fix this throughout the automate.

Reply:
How do I get valid infra data loaded? I have tried manually adding (but typed a random key) as well as `infra_service_load_sample_data` but always get "Could not get cookbooks: invalid org ID: private key block size invalid: private key block size invalid" when going to the org.

Reply:
@msorens pls see the second pinned msg in #automate-infra-views